### PR TITLE
Add scheduling.md for all scheduling settings

### PR DESCRIPTION
**Pull request by boknowswiki**
This document contains all scheduling related settings with explanations.

https://github.com/longhorn/longhorn/issues/1334

Signed-off-by: Bo Tao <bo.tao@rancher.com>

### CODE REVIEW

File: content/docs/0.8.1/volumes-and-nodes/scheduling.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+---
+title: Scheduling
+weight: 8
+---
+
+In this section, you'll learn how longhorn schedules replicas based on multiple factors.
```
Comment (yasker) [inline content/docs/0.8.1/volumes-and-nodes/scheduling.md:6]: `Longhorn`

File: content/docs/0.8.1/volumes-and-nodes/scheduling.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+#### Scheduling Policy
+
+Longhorn scheduling policy has several layers, only the previous layer gets satisfied, the scheduler will goto the next layer, otherwise the scheduling will fail.
```
Comment (yasker) [inline content/docs/0.8.1/volumes-and-nodes/scheduling.md:10]: `several stages`. The word `layer` is a bit weird here.

File: content/docs/0.8.1/volumes-and-nodes/scheduling.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+Then, Longhorn will look for new node with existing zone. And it will schedule the new replica on the new node with existing zone if possible. At this time if `Replica Node Level Soft Anti-Affinity` is un-checked and `Replica Zone Level Soft Anti-Affinity` is checked, this means hard on node but sofe on zone. Longhorn will not schedule the replica if there is no new node with existing zone.
+
+Next, Longhorn will look for existing node with new zone. Practical this scenario should not exist, since one node can only be at one location, it can not have multiple zones. But Longhorn has implemented this logic. At this time if `Replica Node Level Soft Anti-Affinity` is checked and `Replica Zone Level Soft Anti-Affinity` is un-checked, this means soft on node but hard on zone. Longhorn will not schedule the replica if there is existing node with new zone.
```
Comment (yasker) [inline content/docs/0.8.1/volumes-and-nodes/scheduling.md:22]: Remove this paragraph. If it shouldn't exist, it shouldn't be mentioned.

File: content/docs/0.8.1/volumes-and-nodes/scheduling.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+Longhorn scheduling policy has several layers, only the previous layer gets satisfied, the scheduler will goto the next layer, otherwise the scheduling will fail.
+
+1. Node and Zone selection layer, Longhorn will filter the node and zone based on `Replica Node Level Soft Anti-Affinity` and `Replica Zone Level Soft Anti-Affinity`.
+2. Tag selection layer, Longhorn will filter the node which satisfied the first layber with correct tag.
+3. Disk selection layer, Longhorn will filter the disk which satisfied the first two layers based on `Storage Minimal Available Percentage`, `Storage Over Provisioning Percentage` and other disk related factors like request disk space. 
+
+##### Node and Zone Selection Layer
+
+First, Longhorn will always try to schedule the new replica on a new node with a new zone if possible. At this time if both `Replica Node Level Soft Anti-Affinity` and `Replica Zone Level Soft Anti-Affinity` are un-checked, this means hard on node and hard on zone. Longhorn will not schedule the replica if there is no new node with new zone.
```
Comment (yasker) [inline content/docs/0.8.1/volumes-and-nodes/scheduling.md:18]: Don't say `hard on node and hard on zone` (and same below). It's not proper documentation language.

File: content/docs/0.8.1/volumes-and-nodes/scheduling.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+Next, Longhorn will look for existing node with new zone. Practical this scenario should not exist, since one node can only be at one location, it can not have multiple zones. But Longhorn has implemented this logic. At this time if `Replica Node Level Soft Anti-Affinity` is checked and `Replica Zone Level Soft Anti-Affinity` is un-checked, this means soft on node but hard on zone. Longhorn will not schedule the replica if there is existing node with new zone.
+
+Last, Longhorn will looking for existing node with existing zone to schedule the new replica. At this time both `Replica Node Level Soft Anti-Affinity` and `Replica Zone Level Soft Anti-Affinity` should be checked, this means soft on node and soft on zone.
+
+##### Tag Selection Layer and Disk Selection Layer
+
+Once the Node and Zone layer gets satisfied, Longhorn will check the tags on the selected node and the total disk space and avaiable disk space to decide if it can schedule the replica on the disk of the node.
+
+
```
Comment (yasker) [inline content/docs/0.8.1/volumes-and-nodes/scheduling.md:30]: Need more details on `Disk selection stage`.

E.g. how do we decide if a disk is a fit or not? how do we choose from the disks that satisfied all the requirements?

File: content/docs/0.8.1/volumes-and-nodes/scheduling.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+Then, Longhorn will look for new node with existing zone. And it will schedule the new replica on the new node with existing zone if possible. At this time if `Replica Node Level Soft Anti-Affinity` is un-checked and `Replica Zone Level Soft Anti-Affinity` is checked, this means hard on node but sofe on zone. Longhorn will not schedule the replica if there is no new node with existing zone.
+
+Next, Longhorn will look for existing node with new zone. Practical this scenario should not exist, since one node can only be at one location, it can not have multiple zones. But Longhorn has implemented this logic. At this time if `Replica Node Level Soft Anti-Affinity` is checked and `Replica Zone Level Soft Anti-Affinity` is un-checked, this means soft on node but hard on zone. Longhorn will not schedule the replica if there is existing node with new zone.
+
+Last, Longhorn will looking for existing node with existing zone to schedule the new replica. At this time both `Replica Node Level Soft Anti-Affinity` and `Replica Zone Level Soft Anti-Affinity` should be checked, this means soft on node and soft on zone.
+
+##### Tag Selection Layer and Disk Selection Layer
```
Comment (yasker) [inline content/docs/0.8.1/volumes-and-nodes/scheduling.md:26]: Need to make sure the reader understands the disk and node must satisfy all the `tag` specified by the volume in order to be selected.

**Comment by yasker** [comment]
LGTM.

**Comment by catherineluse** [comment]
There are some settings in this PR that I believe should be in the settings reference.

So I'm going to update this PR so that the relevant settings are linked to from the scheduling page, and move the new information to the settings reference.

I'm also reorganizing and updating the settings reference to reflect [the most recent version of the settings](https://github.com/longhorn/longhorn-manager/blob/master/types/setting.go) available from Longhorn Manager. So now the settings reference is structured like this:



    Customizing Default Settings

    General
        Create Default Disk on Labeled Nodes
        Default Data Path
        Default Engine Image
        Default Instance Manager Image
        Enable Upgrade Checker
        Latest Longhorn Version
        Default Replica Count
        Default Longhorn Static StorageClass Name
        Custom Resource API Version
        Automatic Salvage
        Registry Secret
        Volume Attachment Recovery Policy
        Custom mkfs.ext4 parameters

    Backup
        Backup Target
        Backup Target Credential Secret
        Backupstore Poll Interval

    Scheduling
        Replica Node Level Soft Anti-Affinity
        Storage Over Provisioning Percentage
        Storage Minimal Available Percentage
        Disable Scheduling On Cordoned Node
        Replica Zone Level Soft Anti-Affinity

    Danger Zone
        Kubernetes Taint Toleration
        Guaranteed Engine CPU (Experimental)

**Comment by yasker** [comment]
LGTM.

**Pull request merged**